models/cart/src/cart/fields-config.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  Address,
+  CentPrecisionMoney,
+  ClientLogging,
+  Reference,
+} from '@commercetools-test-data/commons';
+import {
+  fake,
+  oneOf,
+  type TModelFieldsConfig,
+  sequence,
+} from '@commercetools-test-data/core';
+import { createRelatedDates } from '@commercetools-test-data/utils';
+import { LineItem } from '../index';
+import {
+  cartState,
+  inventoryMode,
+  origin,
+  shippingMode,
+  taxCalculationMode,
+  taxMode,
+  taxRoundingMode,
+} from './constants';
+import { TCartGraphql, TCartRest } from './types';
+
+const [getOlderDate, getNewerDate] = createRelatedDates();
+
+const commonFieldsConfig = {
```

Comment:
Bear in mind the nested entities are handled differently in REST and GraphQL.
You can take a look at [this section](https://github.com/commercetools/test-data/blob/main/docs/guidelines/creating-new-model.md#handling-references) of the new models docs for details.

Reply:
I will look into this now! Thanks

Reply:
okay i took a look at the docs and this is what I understood, please correct me if this is wrong, cc @CarlosCortizasCT 

**Rest**: has field called store which is actually a reference to a store

**Graphql has 2 fields**
- store which is the actual store
- storeRef which is a reference to a store

Reply:
Thanks for all the details. I have pushed a commit with the new changes to the cart model, please let me know if everything is correct so I can continue with the tests.

The `obj` property is handled at the Reference model level with the transformers 
and I have made all the needed changes for postBuild to ensure that they refs are linked properly 
cc @CarlosCortizasCT

Reply:
I have updated also the cart draft tests and I think now I'm ready for a full review.

Of course the only thing missing now is the sample data which I wasn't able to generate from test-data-generation model, I have left you a slack message about that for whenever you have a moment.